app/src/test/java/org/simple/clinic/security/pin/PinEntryCardControllerTest.kt

```diff
@@ -6,36 +6,64 @@ import com.nhaarman.mockito_kotlin.mock
 import com.nhaarman.mockito_kotlin.times
 import com.nhaarman.mockito_kotlin.verify
 import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Completable
 import io.reactivex.Observable
 import io.reactivex.Single
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.TestScheduler
 import io.reactivex.subjects.PublishSubject
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.After
 import org.junit.Before
 import org.junit.Test
+import org.junit.runner.RunWith
 import org.simple.clinic.patient.PatientMocker
 import org.simple.clinic.security.ComparisonResult
 import org.simple.clinic.security.PasswordHasher
+import org.simple.clinic.security.pin.BruteForceProtection.ProtectedState
+import org.simple.clinic.security.pin.PinEntryCardView.State
 import org.simple.clinic.user.UserSession
 import org.simple.clinic.widgets.UiEvent
+import org.threeten.bp.Clock
+import org.threeten.bp.Duration
+import org.threeten.bp.Instant
+import org.threeten.bp.ZoneOffset.UTC
 
+@RunWith(JUnitParamsRunner::class)
 class PinEntryCardControllerTest {
 
   private val screen = mock<PinEntryCardView>()
   private val userSession = mock<UserSession>()
   private val passwordHasher = mock<PasswordHasher>()
+  private val bruteForceProtection = mock<BruteForceProtection>()
 
   private lateinit var controller: PinEntryCardController
   private val uiEvents = PublishSubject.create<UiEvent>()
-
-  val loggedInUser = PatientMocker.loggedInUser()
+  private val clock = Clock.fixed(Instant.now(), UTC)
```

Comment:
Do you just want to use a `TestClock` here instead of manually managing a clock instance?

Reply:
Also, it won't be a problem in this specific test since we only use it to calculate the differences between two time intervals, but using `Instant.now()` might also introduce flakiness in your tests if you need to verify exact timestamps.

Reply:
Yes!

I keep forgetting that `TestClock()` uses a fixed clock.